src/app/modules/form/form.service.ts

```diff
@@ -174,3 +240,49 @@ export const getFormModelByResponseMode = (
       return EncryptedFormModel
   }
 }
+
+/**
+ * Checks if a form is accessed from within intranet and sets the property accordingly
+ * @param ip The ip of the request
+ * @param publicFormView The form to check
+ * @returns ok(PublicFormView) if the form is accessed from the internet
+ * @returns err(ApplicationError) if an error occured while checking if the ip of the request is from the intranet
+ */
+export const setIsIntranetFormAccess = (
+  ip: string,
+  publicFormView: IPublicFormView,
+): Result<SetRequired<IPublicFormView, 'isIntranetUser'>, ApplicationError> =>
+  IntranetFactory.isIntranetIp(ip).andThen((isIntranetIp) => {
```

Comment:
I'm surprised that this compiles, do you not need a `return` here?

Reply:
ok i think i'll just make the `return` explicit. the outer is an arrow function, so no explicit return is needed. only the inner `andThen` block has curly braces and hence, explicit `return` down below

Reply:
oh sorry misread haha, I didn't notice that there were no curly braces. feel free to ignore this if you haven't already done it